### ISSUE TITLE
omv:version gone?

### ISSUE THREAD

**Issue by mmisw** [mmisw/mmiorr#252]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on May 23, 2010 22:39:52_

What steps will reproduce the problem? 1. request http://mmisw.org/ont/device (or some other; I've tried various
entries with the same behavior)
2. look inside properties of the ontology resource What is the expected output? see an entry for omv:version (among other props) What do you see instead? omv:version is not there

The omv:version should be shown even for an unversioned request. But
omv:version does not show up in any case.

_Original issue: http://code.google.com/p/mmisw/issues/detail?id=252_

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on June 16, 2010 00:47:14_

Fixed, but more internal testing is in progress..

**Labels:** pot

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on June 16, 2010 00:48:20_

New version is now synced in the tree modules (Ont, iserver, orr): 1.9.2.alpha

**Labels:** -pot portal

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on June 16, 2010 00:53:08_

- Fixed for new submissions (new vocabs, mappings, uploads in the various modes--fully-hosted or re-hosted).

- Fixed for unversioned requests, eg., http://mmisw.org/ont/mmi/device - Pending fix for versioned requests, eg., http://mmisw.org/ont/mmi/20100524T062910/device

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on June 16, 2010 10:06:15_

Note, the pending fix mentioned above is for the versioned request when using the direct mechanism, ie, when the URI is "ont"-based: http://mmisw.org/ont/mmi/20100524T062910/device With the 'uri' parameter, the versioned request is properly handled: http://mmisw.org/ont?uri=http://mmisw.org/ont/mmi/20100524T062910/device&form=owl

**Comment by mmisw** [comment]
_From [caru...@gmail.com](https://code.google.com/u/113886747689301365533/) on June 16, 2010 13:08:32_

Fixed.

Ont service version 1.9.3.alpha

Note: as part of the fix, I unified various refactoring attemtps not completed applied. A consequence of this is that content negotiation is now honored where it was not previously. For Firefox in particular, a request with  no explicit format (via file extension or 'form' parameter) will resolve to "html"(*) thus causing a redirection to the ORR portal. Previously, this would resolve in the "owl" response. 

The versioned request example above, but forcing the "owl" response is now properly handled: http://mmisw.org/ont/mmi/20100524T062910/device.owl note the \<omv:version>20100524T062910</omv:version> in the response.

(*) At least with user-agent: [Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10.5; en-US; rv:1.9.1.9) Gecko/20100315 Firefox/3.5.9], the accept list is as follows: [text/html, application/xhtml+xml, application/xml; q = 0.9, */*; q = 0.8], with "text/html" appearing in first place. (By contrast, I see that Safari and Chrome use some "xml" mime type as the dominant entry.)

**Status:** Fixed
